File: python/pangram/pangram.py
"""
pangram provides utility to figure out if a sentence is a pangram.
"""

from string import ascii_lowercase


def is_pangram(sentence: str) -> bool:
    """
    A pangram is a sentence using every letter of the alphabet at least once.
    It is case insensitive.
    A pangram contains each of the 26 letters in the English alphabet.
    """

    return set(sentence.lower()).issuperset(ascii_lowercase)

File: clojure/pangram/src/pangram.clj
(ns pangram (:require [clojure.string :as s]
                      [clojure.set :as set]))

(def ^:private lower-alphabet
  (set (map char (range (int \a) (inc (int \z))))))

(defn pangram? [sentence]
  (set/subset? lower-alphabet (set (s/lower-case sentence))))
